sdk/component/store/local/store.go

```diff
@@ -171,46 +199,90 @@ func (m *manager) Write(ctx context.Context, instanceID uuid.UUID, findings []*o
 
 // Update updates existing vulnerabilities in the underlying database.
 // It returns ErrNoFindingsFound if the passed instanceID is not found.
-func (m *manager) Update(ctx context.Context, instanceID uuid.UUID, findings []*ocsf.VulnerabilityFinding) error {
-	jsonFindings, err := m.marshalFindings(findings)
+func (m *manager) Update(ctx context.Context, instanceID uuid.UUID, findings []*vf.VulnerabilityFinding) error {
+	tx, err := m.db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return errors.Errorf("could not start update transaction: %w", err)
 	}
 
-	stmt, err := m.db.PrepareContext(ctx, `
+	rollback := func(tx *sql.Tx, err error) error {
+		if txErr := tx.Rollback(); txErr != nil {
+			return errors.Errorf("could not rollback transaction for error %w: %w", err, txErr)
+		}
+		return errors.Errorf("unexpected update error, rolled back: %w", err)
+	}
+
+	defer func() {
```

Comment:
why is the rollback deferred over here since there is a rollback lambda that will be invoked if any error is returned?

Reply:
It's a catch all rollback that is a NOOP if the tx was rolledback or committed already

It's just a good safety practice :)